core/src/cg/group4/game_logic/stroll/Stroll.java

```diff
@@ -1,198 +1,269 @@
-     *
-     * @return Subject to subscribe to.
-     */
-    public final Subject getNewEventSubject() {
-        return cNewEventSubject;
-    }
-
-    /**
-     * Getter for the subject to subscribe to to get updated for the end of the event.
-     *
-     * @return Subject to subscribe to.
-     */
-    public final Subject getEndEventSubject() {
-        return cEndEventSubject;
-    }
-}
+package cg.group4.game_logic.stroll;
+
+import cg.group4.game_logic.StandUp;
+import cg.group4.game_logic.stroll.events.StrollEvent;
+import cg.group4.game_logic.stroll.events.TestStrollEvent;
+import cg.group4.game_logic.stroll.events.fishevent.FishingStrollEvent;
+import cg.group4.util.sensors.AccelerationState;
+import cg.group4.util.subscribe.Subject;
+import cg.group4.util.timer.Timer;
+import cg.group4.util.timer.TimerStore;
+import com.badlogic.gdx.Gdx;
+
+import java.util.Observable;
+import java.util.Observer;
+import java.util.Random;
+
+/**
+ * @author Martijn Gribnau
+ * @author Benjamin Los
+ */
+public class Stroll implements Observer {
+
+    /**
+     * The base threshold used for generating events.
+     */
+    protected static final double BASE_THRESHOLD = 0.002;
+
+    /**
+     * Tag used for debugging.
+     */
+    private static final String TAG = Stroll.class.getSimpleName();
+
+
+    protected double cEventThreshold;
+    /**
+     * Amount of rewards collected.
+     */
+    protected int cRewards;
+    /**
+     * Whether or not you're busy with an event.
+     */
+    protected boolean cEventGoing;
+    /**
+     * Whether the stroll has ended or not.
+     */
+    protected Boolean cFinished;
+    /**
+     * The current event being played.
+     */
+    protected StrollEvent cEvent;
+    /**
+     * Subject for end of stroll.
+     */
+    protected Subject cEndStrollSubject;
+    /**
+     * The stroll timer.
+     */
+    protected Timer cStrollTimer;
+
+    /**
+     * The observer to subscribe to the stop subject of stroll timer.
+     */
+    protected Observer cStrollStopObserver = new Observer() {
+
+        @Override
+        public void update(final Observable o, final Object arg) {
+            cFinished = true;
+            if (!cEventGoing) {
+                done();
+            }
+        }
+    };
+
+    protected Observer cUpdateMovementObserver = new Observer() {
+
+        @Override
+        public void update(final Observable o, final Object arg) {
+            cEventThreshold = BASE_THRESHOLD * getAmplifier((AccelerationState) arg);
+        }
+    };
+
+    /**
+     * Subject for new event.
+     */
+    protected Subject cNewEventSubject;
+
+    /**
+     * Subject for end of event.
+     */
+    protected Subject cEndEventSubject;
+
+    /**
+     * Constructor, creates a new Stroll object.
+     */
+    public Stroll() {
+        Gdx.app.log(TAG, "Started new stroll");
+        cRewards = 0;
+        cEventGoing = false;
+        cEventThreshold = BASE_THRESHOLD;

```

Comment:
Shouldn't this be initialized based on the current acceleration status `getAmplifier();